client/src/www/app/main.ts

```diff
@@ -52,12 +51,13 @@ function getRootEl() {
 function createServerRepo(
   eventQueue: EventQueue,
   storage: Storage,
-  deviceSupport: boolean,
-  createTunnel: TunnelFactory
 ) {
-  const repo = new OutlineServerRepository(createTunnel, eventQueue, storage);
-  if (!deviceSupport) {
+  try {
+    return new OutlineServerRepository(eventQueue, storage);
```

Comment:
On your other question, note how we no longer need to know about the tunnel factory here. That is an implementation detail what was leaking to the app logic here.